pkgs/applications/audio/ardour/default.nix

```
@@ -4,146 +4,165 @@
 , aubio
 , boost
 , cairomm
+, cppunit
 , curl
+, dbus
 , doxygen
+, fftw
 , fftwSinglePrec
 , flac
+, fluidsynth
 , glibc
 , glibmm
 , graphviz
 , gtkmm2
+, hidapi
+, itstool
+, libarchive
 , libjack2
 , liblo
+, libltc
 , libogg
+, libpulseaudio
+, librdf_raptor
+, librdf_rasqal
 , libsamplerate
 , libsigcxx
 , libsndfile
 , libusb1
-, fluidsynth_1
-, hidapi
-, libltc
-, qm-dsp
+, libuv
+, libwebsockets
 , libxml2
+, libxslt
 , lilv
 , lrdf
 , lv2
-, makeWrapper
+, pango
 , perl
 , pkg-config
-, itstool
-, python2
+, python3
+, qm-dsp
+, readline
 , rubberband
 , serd
 , sord
 , sratom
+, suil
 , taglib
 , vamp-plugin-sdk
-, dbus
-, fftw
-, pango
-, suil
-, libarchive
 , wafHook
 }:
-let
-  # Ardour git repo uses a mix of annotated and lightweight tags. Annotated
-  # tags are used for MAJOR.MINOR versioning, and lightweight tags are used
-  # in-between; MAJOR.MINOR.REV where REV is the number of commits since the
-  # last annotated tag. A slightly different version string format is needed
-  # for the 'revision' info that is built into the binary; it is the format of
-  # "git describe" when _not_ on an annotated tag(!): MAJOR.MINOR-REV-HASH.
-
-  # Version to build.
-  tag = "5.12";
-in stdenv.mkDerivation rec {
-  name = "ardour-${tag}";
+stdenv.mkDerivation rec {
+  pname = "ardour";
+  version = "6.0";
 
+  # don't fetch releases from the GitHub mirror, they are broken
   src = fetchgit {
     url = "git://git.ardour.org/ardour/ardour.git";
-    rev = "ae0dcdc0c5d13483271065c360e378202d20170a";
-    sha256 = "0mla5lm51ryikc2rrk53max2m7a5ds6i1ai921l2h95wrha45nkr";
+    rev = version;
+    sha256 = "162jd96zahl05fdmjwvpdfjxbhd6ifbav6xqa0vv6rsdl4zk395q";
   };
 
+  patches = [
+    # weird "as" issue: https://tracker.ardour.org/view.php?id=8096
```

```suggestion
    # AS=as in the environment causes build failure https://tracker.ardour.org/view.php?id=8096
```
"AS" set by binutils-wrapper:
https://github.com/NixOS/nixpkgs/blob/023e11ae3407c184eff3463eccc4460d5682a9d1/pkgs/build-support/bintools-wrapper/setup-hook.sh#L57-L66

see similar bugs in other software:
https://bugzilla.mozilla.org/show_bug.cgi?id=1497286
https://bugzilla.mozilla.org/show_bug.cgi?id=1497286

---

@x42 Does the above comment enlighten you as to why the ``as`` issue happens on NixOS and whether the fix should be upstreamed?